### PR TITLE
Add recurring_window maintenance_policy to beta files

### PR DESCRIPTION
**Pull request by gajicdev**
Adding the new beta maintenance_policy "recurring_window" allowing more flexibility to the maintenance policy. The non beta files are not affected.

### CODE REVIEW

File: autogen/main/cluster.tf.tmpl [hunk 1]
```diff
@@ -156,9 +156,27 @@ resource "google_container_cluster" "primary" {
   }
 
   maintenance_policy {
+  {% if beta_cluster %}
+    dynamic "recurring_window"{
+      for_each = local.cluster_maintenance_window_is_recurring
+      content{
```
Comment (morgante) [inline autogen/main/cluster.tf.tmpl]:
```suggestion
      content {
```

File: autogen/main/cluster.tf.tmpl [hunk 2]
```diff
@@ -156,9 +156,27 @@ resource "google_container_cluster" "primary" {
...
+        start_time = var.maintenance_start_time
+        end_time = var.maintenance_end_time
+        recurrence = var.maintenance_recurrence
+      }
+    }
+
+    dynamic "daily_maintenance_window"{
+      for_each = local.cluster_maintenance_window_is_daily
+      content{
```
Comment (morgante) [inline autogen/main/cluster.tf.tmpl]:
```suggestion
      content {
```

File: autogen/main/main.tf.tmpl [hunk 3]
```diff
@@ -164,6 +164,11 @@ locals {
   }]
   # /BETA features
 {% endif %}
+
+{% if beta_cluster %}
+  cluster_maintenance_window_is_recurring = var.maintenance_recurrence != "" && var.maintenance_end_time != "" ? [1] : []
+  cluster_maintenance_window_is_daily = var.maintenance_recurrence != "" && var.maintenance_end_time != "" ? [] : [1]
```
Comment (morgante) [inline autogen/main/main.tf.tmpl]: To make logic easier, let's not repeat the same binary logic.

```suggestion
  cluster_maintenance_window_is_daily = length(local.cluster_maintenance_window_is_recurring) > 0 ? [] : [1]
```

**Comment by gajicdev** [comment]
Thank you morgante!

**Pull request merged**